[DQL3-12] - added deletion_reason field

awset opened the pull request:
https://salsadigital.atlassian.net/browse/DQL3-12

awset committed:
[DQL3-12] - added deletion_reason field

awset committed:
Merge branch 'develop' into feature/DQL3-12-deletion_of_dataset

awset committed:
[DQL3-12] - added automated test and fix other test that is affected

MarkCalvert committed:
Merge branch 'develop' into feature/DQL3-12-deletion_of_dataset

MarkCalvert commented on ckanext/data_qld/dataset_deletion/helpers.py:
```diff
@@ -0,0 +1,40 @@
+import ckan.lib.helpers as h
```
@awset Remember to use the toolkit for helpers etc instead of importing them directly.
Check the toolkit file before importing any ckan core files

MarkCalvert commented on ckanext/data_qld/dataset_deletion/helpers.py:
```diff
@@ -0,0 +1,40 @@
+            h.flash_error('Deletion reason must not less than 10 characters.')
+            return h.redirect_to('/dataset/edit/' + data_dict.id)
+
+    site_user = get_action(u'get_site_user')({u'ignore_auth': True}, {})
```
Why are we using site user instead of the logged-in user?

MarkCalvert committed:
[DQL3-12]
- Updated ckan core imports to use toolkit
- Removed using the site_user as the user for delete reason because we want to record the logged in user who did the delete reason
